### ISSUE TITLE
nginx: lua atpanic: Lua VM crashed, reason: not enough memory

### ISSUE THREAD

**Issue by kollol-chowdhury-incontact** [Kong/kong#8862]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Kong version (`$ kong version`)

2.7.1

### Current Behavior

We are upgrading to Kong 2.7.1 from Kong 2.0.3 and when we are doing that experiencing out of memory error.
We never experienced this issue with 2.0.3 and it has been running perfectly for a long time now.
But right after we started upgrading, started seeing this issue and had to revert back.

We are using C5.2xlarge instances.

Some of the errors are:

malloc(2048) failed (12: Out of memory)
nginx: lua atpanic: Lua VM crashed, reason: not enough memory
malloc(262144) failed (12: Out of memory) while reading response header from upstream

**Comment by fairyqb** [comment]
wei-xin: fairyasky

**Comment by kollol-chowdhury-incontact** [comment]
Hi Team,

Any update on this?

TIA

**Comment by mayocream** [comment]
Hi @kollol-chowdhury-incontact, could you provide more detailed logs and steps to reproduce this issue?

**Comment by kollol-chowdhury-incontact** [comment]
Hi @mayocream , we have not been able to reproduce the issue in lower env and faced this issue twice in prod. We observed that we start getting intermittent 404 error with the message no Route matched with the values. After we do kong reload,it works for like 3mins,and then starts giving the same 404 error. When we checked the logs we found many out of memory error.

We have kong 2.0.3 running without any issue.Facing this with /kong 2.7.1

Also observed that http_allocated_gc has increased drastically in kong 2.7.1 . In kong 2.0.3 it used to be of 0.5 MiB,whereas now it reaches 250MiB . Is this normal?

Facing lots of challenge because of this issue.
